From bd729bf9ab27b2d9e4f4b6a523800cf0744d51e3 Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Thu, 30 May 2019 13:46:21 +0700
Subject: [PATCH 176/248] arm64: dts: renesas: r8a774a1: add Sound SSI device
 node support

This patch adds Renesas R-Car Sound device node and
SSI[0,1,2,3,4,5,6,7,8,9] subnode via Audio-DMAC[0,1] for R8A774A1 SoC.

Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 arch/arm64/boot/dts/renesas/r8a774a1.dtsi | 123 ++++++++++++++++++++++++++++++
 1 file changed, 123 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a774a1.dtsi b/arch/arm64/boot/dts/renesas/r8a774a1.dtsi
index de486b0..dab2ba4 100644
--- a/arch/arm64/boot/dts/renesas/r8a774a1.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a774a1.dtsi
@@ -1664,6 +1664,129 @@
 			#dma-cells = <1>;
 			dma-channels = <16>;
 		};
+
+		rcar_sound: sound@ec500000 {
+			/*
+			 * #sound-dai-cells is required
+			 *
+			 * Single DAI : #sound-dai-cells = <0>; <&rcar_sound>;
+			 * Multi  DAI : #sound-dai-cells = <1>; <&rcar_sound N>;
+			 */
+			/*
+			 * #clock-cells is required for audio_clkout0/1/2/3
+			 *
+			 * clkout       : #clock-cells = <0>;   <&rcar_sound>;
+			 * clkout0/1/2/3: #clock-cells = <1>;   <&rcar_sound N>;
+			 */
+			compatible =	"renesas,rcar_sound-r8a774a1",
+					"renesas,rcar_sound-gen3";
+			reg =	<0 0xec500000 0 0x1000>, /* SCU */
+				<0 0xec5a0000 0 0x100>,  /* ADG */
+				<0 0xec540000 0 0x1000>, /* SSIU */
+				<0 0xec541000 0 0x280>,  /* SSI */
+				<0 0xec740000 0 0x200>;	/*Audio DMAC peri peri*/
+			reg-names = "scu", "adg", "ssiu", "ssi", "audmapp";
+
+			clocks = <&cpg CPG_MOD 1005>,
+				 <&cpg CPG_MOD 1006>, <&cpg CPG_MOD 1007>,
+				 <&cpg CPG_MOD 1008>, <&cpg CPG_MOD 1009>,
+				 <&cpg CPG_MOD 1010>, <&cpg CPG_MOD 1011>,
+				 <&cpg CPG_MOD 1012>, <&cpg CPG_MOD 1013>,
+				 <&cpg CPG_MOD 1014>, <&cpg CPG_MOD 1015>,
+				 <&audio_clk_a>, <&audio_clk_b>,
+				 <&audio_clk_c>,
+				 <&cpg CPG_CORE R8A774A1_CLK_S0D4>;
+			clock-names =	"ssi-all",
+					"ssi.9", "ssi.8", "ssi.7", "ssi.6",
+					"ssi.5", "ssi.4", "ssi.3", "ssi.2",
+					"ssi.1", "ssi.0",
+					"clk_a", "clk_b", "clk_c", "clk_i";
+			power-domains = <&sysc R8A774A1_PD_ALWAYS_ON>;
+			resets = <&cpg 1005>,
+				 <&cpg 1006>, <&cpg 1007>,
+				 <&cpg 1008>, <&cpg 1009>,
+				 <&cpg 1010>, <&cpg 1011>,
+				 <&cpg 1012>, <&cpg 1013>,
+				 <&cpg 1014>, <&cpg 1015>;
+			reset-names =	"ssi-all",
+					"ssi.9", "ssi.8", "ssi.7", "ssi.6",
+					"ssi.5", "ssi.4", "ssi.3", "ssi.2",
+					"ssi.1", "ssi.0";
+			status = "disabled";
+
+			rcar_sound,ssi {
+				ssi0: ssi-0 {
+					interrupts = <GIC_SPI 370
+							IRQ_TYPE_LEVEL_HIGH>;
+					dmas =	<&audma0 0x01>, <&audma1 0x02>,
+						<&audma0 0x15>, <&audma1 0x16>;
+					dma-names = "rx", "tx", "rxu", "txu";
+				};
+				ssi1: ssi-1 {
+					interrupts = <GIC_SPI 371
+							IRQ_TYPE_LEVEL_HIGH>;
+					dmas =	<&audma0 0x03>, <&audma1 0x04>,
+						<&audma0 0x49>, <&audma1 0x4a>;
+					dma-names = "rx", "tx", "rxu", "txu";
+				};
+				ssi2: ssi-2 {
+					interrupts = <GIC_SPI 372
+							IRQ_TYPE_LEVEL_HIGH>;
+					dmas =	<&audma0 0x05>, <&audma1 0x06>,
+						<&audma0 0x63>, <&audma1 0x64>;
+					dma-names = "rx", "tx", "rxu", "txu";
+				};
+				ssi3: ssi-3 {
+					interrupts = <GIC_SPI 373
+							IRQ_TYPE_LEVEL_HIGH>;
+					dmas =	<&audma0 0x07>, <&audma1 0x08>,
+						<&audma0 0x6f>, <&audma1 0x70>;
+					dma-names = "rx", "tx", "rxu", "txu";
+				};
+				ssi4: ssi-4 {
+					interrupts = <GIC_SPI 374
+							IRQ_TYPE_LEVEL_HIGH>;
+					dmas =	<&audma0 0x09>, <&audma1 0x0a>,
+						<&audma0 0x71>, <&audma1 0x72>;
+					dma-names = "rx", "tx", "rxu", "txu";
+				};
+				ssi5: ssi-5 {
+					interrupts = <GIC_SPI 375
+							IRQ_TYPE_LEVEL_HIGH>;
+					dmas =	<&audma0 0x0b>, <&audma1 0x0c>,
+						<&audma0 0x73>, <&audma1 0x74>;
+					dma-names = "rx", "tx", "rxu", "txu";
+				};
+				ssi6: ssi-6 {
+					interrupts = <GIC_SPI 376
+							IRQ_TYPE_LEVEL_HIGH>;
+					dmas =	<&audma0 0x0d>, <&audma1 0x0e>,
+						<&audma0 0x75>, <&audma1 0x76>;
+					dma-names = "rx", "tx", "rxu", "txu";
+				};
+				ssi7: ssi-7 {
+					interrupts = <GIC_SPI 377
+							IRQ_TYPE_LEVEL_HIGH>;
+					dmas =	<&audma0 0x0f>, <&audma1 0x10>,
+						<&audma0 0x79>, <&audma1 0x7a>;
+					dma-names = "rx", "tx", "rxu", "txu";
+				};
+				ssi8: ssi-8 {
+					interrupts = <GIC_SPI 378
+							IRQ_TYPE_LEVEL_HIGH>;
+					dmas =	<&audma0 0x11>, <&audma1 0x12>,
+						<&audma0 0x7b>, <&audma1 0x7c>;
+					dma-names = "rx", "tx", "rxu", "txu";
+				};
+				ssi9: ssi-9 {
+					interrupts = <GIC_SPI 379
+							IRQ_TYPE_LEVEL_HIGH>;
+					dmas =	<&audma0 0x13>, <&audma1 0x14>,
+						<&audma0 0x7d>, <&audma1 0x7e>;
+					dma-names = "rx", "tx", "rxu", "txu";
+				};
+			};
+		};
 	};
 
 	thermal-zones {
-- 
2.7.4

